### PR TITLE
Improve extended metadata controllers

### PR DESCRIPTION
**Pull request by aldafu**
This addresses feature improvements for: https://github.com/aegisbigdata/documentation/issues/131

Changes:

- Support loading and saving for extended metadata pages for dataset and distribution via attached metadata template as RDF string
- Improve form input fields dropdowns with data for license, language and fileformat
- Add date picker for datasets
- Change keyword input field to display input as tags
- Add geographic spatial component for project and datasets
- Include IDs in page URL for dataset and distribution
- Fix to show growl notifications for user interaction

### CODE REVIEW

File: aegis-web/yo/Gruntfile.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ module.exports = function (grunt) {
           livereload: '<%= connect.options.livereload %>'
         },
         files: [
-          '<%= yeoman.app %>/{,*/}*.html',
+          '<%= yeoman.app %>/**/*.html',
```
Comment (fabiankirstein) [inline aegis-web/yo/Gruntfile.js:118]: Why is that change needed?

File: aegis-web/yo/app/index.html [hunk 2]
```diff
@@ -102,9 +103,11 @@
     <link rel="stylesheet" href="styles/jupyter-tool.css">
     <link rel="stylesheet" href="styles/tools-jupyter-overview.css">
     <link rel="stylesheet" href="styles/leaflet-areaselect.css">
+    <link rel="stylesheet" href="styles/ng-tags.custom.css">
     <!-- endbuild -->
   </head>
   <body  ng-app="hopsWorksApp">
+    <div growl></div>
```
Comment (fabiankirstein) [inline aegis-web/yo/app/index.html:110]: Just curious. what is that?

File: aegis-web/yo/app/scripts/controllers/extendedMetadataCtrl.js [hunk 3]
```diff
@@ -39,7 +39,8 @@
 
 'use strict';
 
-
+const AEGIS_PROJECT_TEMPLATE_ID = 14;
```
Comment (fabiankirstein) [inline aegis-web/yo/app/scripts/controllers/extendedMetadataCtrl.js:42]: I'm still not sure if we should use only compliant JavaScript.

File: aegis-web/yo/app/scripts/directives/extmdHeader.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+ 'use strict';
+
+ angular.module('hopsWorksApp').directive('extmdHeader', function() {
```
Comment (fabiankirstein) [inline aegis-web/yo/app/scripts/directives/extmdHeader.js:3]: Can we name this more general? So we can extend it later to the header directive everybody uses.

File: aegis-web/yo/Gruntfile.js (hunk 1 above)
Comment (aldafu) [reply aegis-web/yo/Gruntfile.js:118]: So that the browser will reload the page once any HTML file is saved. Without this change the page has to be reloaded manually to see changes.

File: aegis-web/yo/app/index.html (hunk 2 above)
Comment (aldafu) [reply aegis-web/yo/app/index.html:110]: This is needed to display popup notifications to the user, e.g. whether saving form data was successfully or failed. The code already used the growl library for that, but notifications would not display without this required div.

File: aegis-web/yo/Gruntfile.js (hunk 1 above)
Comment (cbotsikas) [reply aegis-web/yo/Gruntfile.js:118]: I was also curious, according to [globbing patterns](https://gruntjs.com/configuring-tasks#globbing-patterns), it looks correct. I think `{,*/}` was a poor implementation of `**/`

File: aegis-web/yo/app/scripts/controllers/extendedMetadataCtrl.js (hunk 3 above)
Comment (aldafu) [reply aegis-web/yo/app/scripts/controllers/extendedMetadataCtrl.js:42]: `const` is already used in elsewhere `mainCtrl.js`. I tested with `grunt build` and got no JS errors in the browser. All browser versions since 2016 support `const` and `let`, see https://caniuse.com/#feat=const

But if prefered we can changed to `var`.

File: aegis-web/yo/app/scripts/directives/extmdHeader.js (hunk 4 above)
Comment (aldafu) [reply aegis-web/yo/app/scripts/directives/extmdHeader.js:3]: Yes, as discussed. Either we merge this now as initial version for the header and improve on that and change the name in further PRs or we can remove it from this PR.

**Review by fabiankirstein — APPROVED** [review]
It's fine for me.

**Comment by fabiankirstein** [comment]
@cbotsikas  Can you also check this?

File: aegis-web/yo/app/scripts/directives/extmdHeader.js (hunk 4 above)
Comment (cbotsikas) [reply aegis-web/yo/app/scripts/directives/extmdHeader.js:3]: I think it's ok but we need a general one too that will include in the template both gray and white header parts of the current template and transclude the white part. breadcrumb should be part of it and/or be configurable via attributes.

**Review by cbotsikas — APPROVED** [review]
LGTM
